cmake/modules/BuildBoost.cmake

```diff
@@ -135,7 +135,10 @@ function(do_build_boost root_dir version)
     list(APPEND b2 binary-format=elf)
   endif()
   if(WITH_BOOST_VALGRIND)
+    message(STATUS "BUILDING Boost Libraries: WITH_BOOST_VALGRIND is set to ON")
     list(APPEND b2 valgrind=on)
+  else()
+    message(STATUS "BUILDING Boost Libraries: WITH_BOOST_VALGRIND is set to OFF")
```

Comment:
I’d suggest drop this commit. These logging messages do help with debugging, but they should not get merged.

Reply:
but isn't that dependent on the environment (whether boost is installed or not)?
in this case it is probably good to keep the log messages.

Reply:
ack and ack - removed the OFF log messages to lower the log noise, the ON logs are useful for analyzing previous Shaman build logs in hindsight from the time that the boost Valgrind issues started occurring (this has happened several times)

Reply:
unlike the logging in python or in C++ in Ceph, where we tries to understand the runtime behavior, the cmake options are explicitly passed to cmake when generating building system. so we can always understand its values by looking at the command line argument and its default value.

so, IMHO, these logging message help us to understand how CMake works, but they do not help when it comes to the behavior of our own CMake script.

Reply:
> but isn't that dependent on the environment (whether boost is installed or not)?
> in this case it is probably good to keep the log messages.

please note `WITH_SYSTEM_BOOST` is specified explicitly, so if it's not found. the generation phase fails. it's determined.

Reply:
my concern was with cases determined at compile time automatically. e.g.
https://github.com/ceph/ceph/blob/main/src/script/run-make.sh#L44

but this would be reflected in the cmake commandline, right?

Reply:
> but this would be reflected in the cmake commandline, right?

yes, that's exactly my point.

Reply:
The intent of adding the log messages is operational. Problems with Valgrind have recurred several times before sporadically as a result of environmental changes and not CMake code changes, logs would assist in shortening the diagnostic cycles.